src/app/data-model-provider/MetadataTypes.h

```diff
@@ -84,18 +85,102 @@ enum class AttributeQualityFlags : uint32_t
     kTimed           = 0x0040, // `T` quality on attributes (writes require timed interactions)
 };
 
+
+struct attribute_entry_mask_t {
```

Comment:
this seems to be private. Place it inside the AttributeEntry or into some `internal` or `detail` namespace. We want to be very clear that this is not a public API.